### PR TITLE
thumbnail : ensure we get the right size, avoiding rounding errors

### LINKED ISSUE #9003: (minor) thumbnails misaligned by 1 pixel in filmstrip depending on aspect ratio
**Describe the bug/issue**

This is really nitpicking here, so I would understand if this issue was closed with a "won't do". 

Anyway, thumbnail pictures are not perfectly centered in their 'slot' in the filmstrip. The left border is 1 pixel too wide (or the right one is 1 pixel too thin, depending on your point of view). Since my pictures have a 3:2 aspect ratio and vertical/square pictures are not affected, I assume that this occurs because the pixel width of the 'box' is always the same and therefore thumbnails with an even or uneven pixel number can not be centered properly.

A fix would probably require to crop a single pixel row from the thumbnail to make the count always even/uneven or by making the width of the box one pixel wider/smaller depending on the shape of the thumbnail. 

To reproduce, use this CSS snippet to better visualize the area around a picture. My screen has a 1680/1080 resolution - other resolutions might require different values.
```
#thumb_main:active #thumb_back
{
  border: 4px solid shade(@thumbnail_selected_bg_color, 1.25);
}
```
image
As you can see, the thumbnail overlaps on the right side but not on the left side.

image
Without the white border it is really hard to spot, but the splitting line is not perfectly centered.

Like I said, it is nitpicking, but once you see it, you cannot unsee it ;)

Manjaro Linux - 3.5.0+1967~g4024d7d73

### PR DESCRIPTION
**Pull request by AlicVB**
this fix #9003 

@xfh @Solarer can you confirm it fix it for you too ? and don't produce unwanted artefacts ?
Thanks !

### CODE REVIEW

**Comment by xfh** [comment]
I can confirm, that works. That was fast, thanks Alic!

File: src/views/view.c [hunk 1]
```diff
@@ -1019,9 +1019,11 @@ dt_view_surface_value_t dt_view_image_get_surface(int imgid, int width, int heig
   }
 
   // so we create a new image surface to return
-  const float scale = fminf(width / (float)buf_wd, height / (float)buf_ht) * darktable.gui->ppd_thb;
-  const int img_width = buf_wd * scale;
-  const int img_height = buf_ht * scale;
+  float scale = fminf(width / (float)buf_wd, height / (float)buf_ht) * darktable.gui->ppd_thb;
+  const int img_width = buf_wd * scale + 0.5f;
```
Comment (TurboGit) [inline src/views/view.c:1023]: isn't this equivalent to : 

const int img_width = roundf((float)buf_wd * scale);

?

Likewise below.

Comment (AlicVB) [reply src/views/view.c]: yep, you're right, and roundf is more explicit ! Done

Comment (Solarer) [reply src/views/view.c]: Almost fixed for me.
This fixes the alignment in lightable culling mode, but in darktable it behaves weird:
When I open an image for the first time in darkroom, the thumbnail is displayed correctly while darktable reports: "working loading IMG_xxx"
Then, darktable displays: "working" and this misaligns the picture again by 1 pixel

**Review by TurboGit — APPROVED** [review]
Thanks !

File: src/views/view.c (hunk 1 above)
Comment (AlicVB) [reply src/views/view.c]: oops... I've forgot that case. PR #9117 should fix it.
Thanks for the tests !

**Pull request merged**